### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [mozilla/rr#1545]
Would you like to add more error handling for return values from functions like the following?
- [close](http://pubs.opengroup.org/onlinepubs/9699919799/functions/close.html "Close a file descriptor.") ⇒ [AutoRemoteSyscalls::retrieve_fd_arch](https://github.com/mozilla/rr/blob/bad1d75f86e68aace5cce61d7eced4f1e9b707da/src/AutoRemoteSyscalls.cc#L397)
- [fprintf](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fprintf.html "Print formatted output.") ⇒ [Command::print_help_all](https://github.com/mozilla/rr/blob/bfcfc0d392c73bf70b98873bc581abc202085a48/src/Command.cc#L46)

**Comment by rocallahan** [comment]
`close`: yes. And `unlink` too.
`fprintf`: no. We assume stdio can't fail.
`pthread_create`: no, it's not that important to have error coverage in tests.

**Comment by elfring** [comment]
I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Unchecked return values") a bit more.
Would you like to detect every error situation as early as possible?

**Comment by rocallahan** [comment]
No. It often adds clutter to the code without providing value. For example, in my entire career I have never observed `printf` return a failure value, and adding error checking of `printf`s would make code significantly less readable.

**Comment by elfring** [comment]
Do you care if the output target can [become full](https://en.wikipedia.org/wiki//dev/full "Checking storage allocation")?

How do you think about to improve [static source code analysis](https://en.wikipedia.org/wiki/List_of_tools_for_static_code_analysis#C.2FC.2B.2B "Tools for static code analysis") also for your software?
